### PR TITLE
[WIP] Replace  Shortcut With KeyHolder

### PR DESCRIPTION
**Pull request by AkaShark**

### CODE REVIEW

**Comment by AkaShark** [comment]
I want to split GeneralTab.swift into some Views such as 
CleanShot 2024-01-18 at 09 44 52@2x
CleanShot 2024-01-18 at 09 48 51@2x
but I am not familiar with swfitUI if there had any hidden dangers

**Comment by CanglongCl** [comment]
> I want to split GeneralTab.swift into some Views such as CleanShot 2024-01-18 at 09 44 52@2x CleanShot 2024-01-18 at 09 48 51@2x but I am not familiar with swfitUI if there had any hidden dangers

This is a good practice and is also recommended.

File: Easydict/NewApp/View/SettingView/Tabs/View/Genearl/GeneralOtherSetting.swift [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+                    Toggle(isOn: $enableBetaNewApp) {
+                        Text("enable_beta_new_app")
+                    }
+                }
+            } header: {
+                Text("other")
+            }
+        }
+
+        @AppStorage(kHideMainWindowKey) private var hideMainWindow = false
```
Comment (CanglongCl) [inline Easydict/NewApp/View/SettingView/Tabs/View/Genearl/GeneralOtherSetting.swift:45]: We can replace this with `@Default(.hideMainWindow) private var hideMainWindow`.

**Review by CanglongCl — CHANGES_REQUESTED** [review]

File: Easydict/NewApp/View/SettingView/Tabs/View/Genearl/GeneralOtherSetting.swift (hunk 1 above)
Comment (AkaShark) [reply Easydict/NewApp/View/SettingView/Tabs/View/Genearl/GeneralOtherSetting.swift:45]: okay I  got it

**Comment by tisfeng** [comment]
@AkaShark  Please don't push branch to main repository directly, you should push branch `shortcut` to your forked repo, and then open a PR, like this https://github.com/tisfeng/Easydict/pull/335

image

image

**Comment by AkaShark** [comment]
> @AkaShark Please don't push branch to main repository directly, you should push branch `shortcut` to your forked repo, and then open a PR, like this #335
> 
> image image

okay sorry, I forget it... I will change the commit into my forked repo

**Comment by tisfeng** [comment]
@AkaShark Please update code, move shortcuts code in Genearl to `GeneralTab.swift` and `MenuItemView.swift`, we need to test it in real use.

**Comment by AkaShark** [comment]
yeah I know it, I will open a new PR later

**Pull request closed without merging**